http-mclient/src/Network/HTTP/MClient.hs

```diff
@@ -138,16 +141,44 @@ send' transactionDecoder decoder request = do
   sendRequest sendRequest' transactionDecoder decoder request
 
 sendRequest
-  :: SendRequest
+  :: forall env a . Env env
+  => SendRequest
   -> TransactionDecoder a
   -> ResponseDecoder a
   -> HTTP.Request
   -> MIO env (Result a)
 sendRequest sendRequest' transactionDecoder decoder request =
-  either transactionDecoder decoder
-    <$> Exception.tryJust selectException (liftIO $ sendRequest' request)
+  either transactionDecoder decoder <$> retriableSendRequest
   where
-    selectException :: HTTP.HttpException -> Maybe HTTP.HttpExceptionContent
-    selectException = \case
-      (HTTP.HttpExceptionRequest _request content) -> pure content
-      _other                                       -> empty
+    retriableSendRequest :: MIO env (Either HTTP.HttpExceptionContent (HTTP.Response LBS.ByteString))
+    retriableSendRequest = Retry.retrying limitedBackoff (const isRetriable) process
+      where
+        process
+          :: Retry.RetryStatus
+          -> MIO env (Either HTTP.HttpExceptionContent (HTTP.Response LBS.ByteString))
+        process Retry.RetryStatus{..} = do
+          when (rsIterNumber > 0)
+            . Log.info
+            $ "Retrying failed request on attempt: " <> convert (show rsIterNumber)
```

Comment:
__question:__ Not a blocker, but is it possible to also log what kind of failure the request was?

If it's easy then it's probably good information to add. A request can fail for 4 reasons, and it might be useful to distinguish between them when logging.

Reply:
Not a blocker, but is it possible to also log what kind of failure the request was?

@dkubb I am going to go ahead and log the failures.